infrastructure/network/connmanager/outgoing_connections.go

```diff
@@ -42,7 +42,6 @@ func (c *ConnectionManager) checkOutgoingConnections(connSet connectionSet) {
 		err := c.initiateConnection(addressString)
 		if err != nil {
 			log.Infof("Couldn't connect to %s: %s", addressString, err)
-			c.addressManager.RemoveAddress(netAddress)
```

Comment:
If you don't remove the address here, doesn't it mean that we keep all addresses indefinitely?